clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10402,8 +10402,7 @@ def err_builtin_launder_invalid_arg : Error<
 def err_sycl_attribute_address_space_invalid : Error<
   "address space is outside the valid range of values">;
 def err_sycl_kernel_name_class_not_top_level : Error<
- "kernel name class and its template argument classes' declarations can only "
- "nest in a namespace: %0">;
+ "kernel needs to have a globally-visible name">;
```

Comment:
Hi @AlexeySachkov , 
How about switching the message to be `"kernel name missing or not globally accessible type name"` ?

I also have a PR for this, but yours is more thorough. I'll close mine.

Reply:
Good suggestion, done in 0456e58562c4e035b7b5d1b526cf7fbee37b552c